cmd/ddev/cmd/composer-createproject.go

```diff
@@ -0,0 +1,129 @@
+package cmd
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/drud/ddev/pkg/ddevapp"
+	"github.com/drud/ddev/pkg/output"
+	"github.com/drud/ddev/pkg/util"
+	"github.com/spf13/cobra"
+)
+
+var ComposerCreateProjectCmd = &cobra.Command{
+	Use:   "create-project [options] [package] [directory] [version]",
+	Short: "Executes 'composer create-project' within the web container",
+	Long: `Directs the use of 'composer create-project' within the context of the web
+container. Projects will be installed to a temporary directory and moved to the
+project root directory after installation. Any existing files in the project
+root will be deleted when creating a project.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// Separate positional arguments from flags. Positional arguments will be manipulated
+		// to preserve the optionality of the package, directory, and version arguments, and
+		// flags will be collected and passed to the command unmodified.
+		positionals := make([]string, 0)
+		flags := make([]string, 0)
+		for _, arg := range args {
+			if strings.HasPrefix(arg, "-") {
+				flags = append(flags, arg)
+				continue
+			}
+
+			positionals = append(positionals, arg)
+		}
+
+		// Package, directory, and version arguments are optional, positional, and difficult to parse.
+		// This will capture the possible argument combinations and assign them appropriately.
+		var pack, dir, ver string
+		if len(positionals) >= 3 {
+			pack = positionals[len(positionals)-3]
+			dir = positionals[len(positionals)-2]
+			ver = positionals[len(positionals)-1]
+		} else if len(positionals) == 2 {
+			pack = positionals[len(positionals)-2]
+			dir = positionals[len(positionals)-1]
+		} else if len(positionals) == 1 {
+			pack = positionals[len(positionals)-1]
+		}
+
+		app, err := ddevapp.GetActiveApp("")
+		if err != nil {
+			util.Failed(err.Error())
+		}
+
+		// Ensure project is running
+		if app.SiteStatus() != ddevapp.SiteRunning {
+			err = app.Start()
+			if err != nil {
+				util.Failed("Failed to start app %s to run create-project: %v", app.Name, err)
+			}
+		}
+
+		// Make the user confirm that existing contents will be deleted
+		// If the user hasn't passed any positional arguments, we're not actually executing an
+		// install and nothing will be deleted -- for example, 'ddev composer create-project -h'
+		doInstall := false
+		if len(positionals) > 0 {
+			util.Warning("Warning: Any existing contents of the project root will be overwritten")
```

Comment:
I noticed this warning; it's not necessarily true is it? Depending on whether they used the all-powerful "." as the destination (which is probably what they want). Maybe it needs to get the path of what it's going to blow away and provide that in the warning?

Reply:
Well, it's true as far as whatever exists in `/var/www/html` will be explicitly `rm`ed before continuing with the installation. The catch, as I think you're pointing out here, is if a user provided an absolute path to the installation directory, then there's really no reason to delete the project root. At the very least this needs the path of what's being deleted.